### PR TITLE
Ripping and tearing

### PR DESCRIPTION
**Pull request by matthuhiggins**
Delete all the things
@malept @ebarendt

### CODE REVIEW

File: lib/slither/column.rb [hunk 1]
```diff
@@ -1,114 +1,41 @@
...
-      @truncate = options[:truncate] || false
-      # Only used with floats, this determines the decimal places
-      @precision = options[:precision]
     end
 
     def unpacker
       "A#{@length}"
     end
 
-    def parse(value)
```
Comment (matthuhiggins) [inline lib/slither/column.rb:26]: This created a new string for everything.

File: lib/slither/section.rb [hunk 2]
```diff
@@ -1,103 +0,0 @@
...
-
-    private
-
-      def unpacker
-        @unpacker ||= @columns.map { |c| c.unpacker }.join('')
-      end
-
-      def divide(string, sections)
-        result = []
-        str = string.dup
```
Comment (matthuhiggins) [inline lib/slither/section.rb:88]: This created a new string for no reason.

File: lib/slither/section.rb [hunk 3]
```diff
@@ -1,103 +0,0 @@
...
-        unless @definition.options[:force_character_offset]
-          result = str.unpack(unpacker)
-          result.each do |s|
-            s.force_encoding(string.encoding) if s.respond_to? :force_encoding
-          end
-        else
-          sections.each do |s|
-            result << str.slice!(0..(s-1))
-          end
```
Comment (matthuhiggins) [inline lib/slither/section.rb:97]: This code path was never used, but it seems more sane to me

Comment (matthuhiggins) [inline lib/slither/section.rb:97]: We should investigate if this is an option.

File: lib/slither/column.rb [hunk 4]
```diff
@@ -1,114 +1,41 @@
...
       @options = options
-      @alignment = options[:align] || :right
-      @type = options[:type] || :string
-      @padding = options[:padding] || :space
-      @truncate = options[:truncate] || false
-      # Only used with floats, this determines the decimal places
-      @precision = options[:precision]
     end
 
     def unpacker
```
Comment (matthuhiggins) [inline lib/slither/column.rb:13]: This created a new string for every line.

File: lib/slither/section.rb [hunk 5]
```diff
@@ -1,103 +0,0 @@
...
-      # ) unless @columns.size == data.size
-      row = ''
-      @columns.each do |column|
-        row += column.format(data[column.name])
-      end
-      row
-    end
-
-    def parse(line)
-      line_data = divide( line, @columns.map(&:length) )
```
Comment (matthuhiggins) [inline lib/slither/section.rb:51]: This created a new array for every line, with the call to map.

**Comment by ebarendt** [comment]
Awesome. Yeah, String#slice would be good. Not sure why we wouldn't be able to use that.

**Pull request merged**